ouroboros-consensus/src/Ouroboros/Consensus/Node/Recovery.hs

```diff
@@ -83,6 +71,67 @@ exceptionRequiresRecovery pb e = case toExitReason pb e of
     DatabaseCorruption -> True
     _                  -> False
 
+-- | A bracket function that manages the clean-shutdown marker on disk.
```

Comment:
Oh, wait, we have duplicated comments. Why not keeping this, and explaining what are the 

```
LastShutDownWasClean
```
and 

```
ChainDB m blk -> m a -> m a
```

parameters in the `llrnWithCheckedDB` constructor?

Reply:
Done